data-app/db/Models.py

```diff
@@ -132,7 +132,28 @@ def __repr__(self) -> str:
         return retVal
 
 
-engine = create_engine("sqlite:///../ProcessedData.db",echo=False,future=True)
+engine = create_engine(
+    "sqlite:///../ProcessedData.db",
+    echo=False,
+    future=True,
+    pool_pre_ping=True,
+    pool_recycle=300,
+    connect_args={
+        'timeout': 30,
+        'check_same_thread': False
+    }
+)
+
+# Enable WAL mode for better concurrent access
+@event.listens_for(engine, "connect")
+def set_sqlite_pragma(dbapi_connection, connection_record):
+    cursor = dbapi_connection.cursor()
+    cursor.execute("PRAGMA journal_mode=WAL")
+    cursor.execute("PRAGMA synchronous=NORMAL")
+    cursor.execute("PRAGMA temp_store=MEMORY")
+    cursor.execute("PRAGMA mmap_size=134217728")  # 128MB
+    cursor.close()
+
```

Comment:
Why are you changing the database settings on connect? We don't need a Write Ahead Log or any of these settings by default.

What caused this to be necessary?